### PR TITLE
fix: teacherView show deadline

### PR DESCRIPTION
**Pull request by solaolateju**
Deadline: time and message visible in teacher view
Deadline - Google Chrome 2020-08-01 13 41 56

### CODE REVIEW

File: src/components/modes/student/StudentView.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const StudentView = ({ t, deadlineMessage, initialDateTime }) => {
   let dateTimeInMille = new Date(dateTimeLocal) - new Date();
   const isElapsed = dateTimeInMille < 0;
   const pendingDeadlineConfigMsg =
-    'You do not have a deadline to complete this activity.';
+    'You do not have a completion deadline this lesson.';
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentView.js:33]: Typos. Missing "for".

Comment (juancarlosfarah) [reply src/components/modes/student/StudentView.js:33]: Also, use `t`.

File: src/components/modes/teacher/Settings.js [hunk 2]
```diff
@@ -129,6 +129,17 @@ class Settings extends Component {
             label={t('Show Header to Students')}
           />
         </Grid>
+        <Grid item>
+          <TextField
+            id="timeoutMessage"
+            label="Timeout message"
+            placeholder="Show message when timer runs out"
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:136]: Use `t`. "Message to show when timer runs out"

File: src/components/modes/teacher/Settings.js [hunk 3]
```diff
@@ -129,6 +129,17 @@ class Settings extends Component {
             label={t('Show Header to Students')}
           />
         </Grid>
+        <Grid item>
+          <TextField
+            id="timeoutMessage"
+            label="Timeout message"
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:135]: Use `t`. "Timeout Message".

File: src/components/modes/teacher/TeacherView.js [hunk 4]
```diff
@@ -37,13 +33,9 @@ export class TeacherView extends Component {
       message: PropTypes.string,
       fab: PropTypes.string,
     }).isRequired,
-    dispatchGetUsers: PropTypes.func.isRequired,
+    dispatchPatchAppInstance: PropTypes.func.isRequired,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:36]: You don't need to `getUsers` anymore?

File: src/components/modes/teacher/TeacherView.js [hunk 5]
```diff
@@ -150,23 +151,20 @@ export class TeacherView extends Component {
...
 });
 
 // allow this component to dispatch a post
 // request to create an app instance resource
 const mapDispatchToProps = {
-  dispatchGetUsers: getUsers,
-  dispatchPostAppInstanceResource: postAppInstanceResource,
-  dispatchPatchAppInstanceResource: patchAppInstanceResource,
-  dispatchDeleteAppInstanceResource: deleteAppInstanceResource,
+  // dispatchPostAppInstance: postAppInstance,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:165]: How come you don't need this anymore?

File: src/reducers/appInstance.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ import { showErrorToast } from '../utils/toasts';
 
 const DEFAULT_SETTINGS = {
   headerVisible: false,
+  deadlineMessage: 'Time is up! ',
```
Comment (juancarlosfarah) [inline src/reducers/appInstance.js:14]: Translation?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -10,12 +10,16 @@
   "homepage": ".",
   "license": "MIT",
   "dependencies": {
+    "@date-io/date-fns": "2.10.8",
+    "@date-io/moment": "1.x",
```
Comment (juancarlosfarah) [inline package.json:14]: Pin exact version.

File: src/components/modes/teacher/Settings.js [hunk 8]
```diff
@@ -51,19 +60,43 @@ class Settings extends Component {
...
   state = {
-    selectedDate: new Date(),
+    selectedDate: null,
   };
 
+  componentDidMount() {
+    const { lang } = this.props;
+    // import corresponding moment locale depending on language
+    // eslint-disable-next-line no-unused-expressions
+    import(`moment/locale/${lang}`);
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:82]: Don't we need to run this again in `componentDidUpdate`?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! Just some minor questions / comments.

File: package.json (hunk 7 above)
Comment (pyphilia) [reply package.json:14]: In the installation guide, it is required to use this version for date-pickers https://material-ui-pickers.dev/getting-started/installation
```For material-ui-pickers v3 use v1.x version of @date-io adapters.```

**Pull request merged**